plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -857,9 +857,9 @@ class FossId internal constructor(
         }
 
         val matchedLines = mutableMapOf<Int, MatchedLines>()
-        val snippets = runBlocking(Dispatchers.IO) {
-            pendingFiles.map {
-                async {
+        val snippets = pendingFiles.associateWith {
+            val filteredSnippets = lazy {
```

Comment:
Instead of having lazy sets of snippets, could this be solved with sequences of snipptes instead of maybe simplify the code a bit?

Reply:
Could you please suggest with a code snippet how to achieve that ?
Would it be as simple as:
```
sequence{
  yieldAll(querySnippets())
}
``` 
? Would the same objective to lazy load the snippets be achieved?

Reply:
Probably anyway the right way is to stick with `async`, but make it [start lazily](https://kotlinlang.org/docs/composing-suspending-functions.html#lazily-started-async).

Reply:
So it means the `snippets` in the `RawResults` are now `Map<String, Deferred<MutableSet<Snippet>>>` ?

Reply:
I guess something like `MutableSet<Deferred<Snippet>>` would make more sense as values, because that allows to call `awaitAll()` on the values. But I don't see through the required changes right now; you'd have to find our yourself 😉

Reply:
If the strategy is to process files one by one until the limit is reached, the `async` is probably a bit pointless, since actually the requests are executed sequentially. It would probably be possible to still run things in parallel (maybe with a lower number of concurrent jobs) and cancelling jobs early when the limit has already been reached; in the worst case, more results than configured would be fetched and ignored afterwards. But I fear that this approach does not work well with the current separation between listing snippets and mapping them/applying choices.

Reply:
Ok, it was just an untested idea anyway. Use whatever solution you see fit.